slurm/memory-hog.py

```diff
@@ -1,42 +1,57 @@
 #!/usr/bin/env python3
-from __future__ import print_function, division
 import argparse
-import random
-from random import uniform
-import sys
+from time import sleep
+import gc
+import platform
+import resource
 
-parser = argparse.ArgumentParser()
-parser.add_argument('mem', help="Use this much memory")
-args = parser.parse_args()
+if __name__ == "__main__":
+      
+    parser = argparse.ArgumentParser()
+    parser.add_argument('mem', metavar="memory", 
+            help="Use this much memory")
+    parser.add_argument('--sleep', 
+            help="Sleep this many seconds", type=int)
+    args = parser.parse_args()
 
+    # calculate the amount of memory requested in bytes
+    mem = args.mem.lower()
 
-mem = args.mem
-if mem.endswith('b'):            mem = int(mem[:-1])
-elif mem.lower().endswith('k'):  mem = int(mem[:-1]) * 1000**1
-elif mem.lower().endswith('m'):  mem = int(mem[:-1]) * 1000**2
-elif mem.lower().endswith('g'):  mem = int(mem[:-1]) * 1000**3
-elif mem.lower().endswith('t'):  mem = int(mem[:-1]) * 1000**4
-else:                            mem = int(mem)
+    if mem.endswith('b'):
+        mem = int(mem[:-1])
 
-try:
-    range = xrange
-except NameError:
-    pass
```

Comment:
I think this was another python2 thing...

Reply:
Yes but it's not used in the code anyway.

Reply:
+1